### PR TITLE
Continuation of the object model

### PR DESCRIPTION
**Pull request by lvdd-rbl**
- Fixed issue with OrderMachine : a customer can have one or more demands but a demand is only linked to a single type of machine. The relation is updated in order to have one type of machine for each demand (a machine can be associated with multiple demands).
- The text file instance includes for each technician a list of machines they can install (0 or 1). As a result, I decided to create an association entity (Installation) that describes the relationship between a machine and a technician.
- Some quick updates (no setters for ID, GenerationType.IDENTITY instead of AUTO, redefined equals/hashCode/toString, etc)
- Added Instance entity
- Added Planning entity
- Added attributes to Vehicle and Technician (general attributes given in the header of each instance in the text files)
- Added @MapKey annotation in Point for the myRoutes attribute
- Added a class to test the object model (with a quick test of the Scanner object that is able to read the text file)
- A bit of JavaDoc
- Some clear() methods here and there that will be useful later for the optimization part
- Etc

### CODE REVIEW

**Review by LouisCalypso — APPROVED** [review]

**Review by LouisCalypso — APPROVED** [review]

**Review by LouisCalypso — APPROVED** [review]

File: src/modele/Customer.java [hunk 1]
```diff
@@ -40,58 +34,81 @@
...
     private Set<Demand> customerDemands;
-
+    
+    /**
+     * No-argument constructor
+     */
     public Customer() {
         super();
         this.position = 0;
-        this.customerDemands = new HashSet<Demand>();
```
Comment (Osgilia) [inline src/modele/Customer.java:56]: why ?

File: src/modele/Customer.java [hunk 2]
```diff
@@ -40,58 +34,81 @@
...
-        super(2, x, y);
+    /**
+     * Parameterized constructor
+     *
+     * @param id
+     * @param x
+     * @param y
+     * @param demand
+     */
+    public Customer(Integer id, double x, double y, int demand) {
```
Comment (Osgilia) [inline src/modele/Customer.java:61]: mais l'id est généré automùatiquement, non ?

File: src/modele/Depot.java [hunk 3]
```diff
@@ -29,50 +25,56 @@
...
-
-    public Long getId() {
-        return id;
+        this.vehicleSet = new HashSet<>();
     }
 
-    public void setId(Long id) {
-        this.id = id;
+    public Depot(Integer id, double x, double y) {
+        super(id, 1, x, y);
```
Comment (Osgilia) [inline src/modele/Depot.java:37]: pk id ?

File: src/modele/Point.java [hunk 4]
```diff
@@ -10,57 +10,92 @@
...
         this.x = 0;
         this.y = 0;
-
+        this.myRoutes = new HashMap<>();
+        this.collectionVehicules = new HashSet<>();
     }
 
-    public Point(Integer pointType, double x, double y) {
+    public Point(Integer id, Integer pointType, double x, double y) {
         this();
```
Comment (Osgilia) [inline src/modele/Point.java:82]: ?

File: src/modele/Customer.java (hunk 1 above)
Comment (lvdd-rbl) [reply src/modele/Customer.java:56]: still in the code, I don't understand why it's in red

File: src/modele/Customer.java (hunk 2 above)
Comment (lvdd-rbl) [reply src/modele/Customer.java:61]: id are given in the text files
it will be best to follow them, because locations are associated with ids

**Review by LouisCalypso — APPROVED** [review]

**Review by Osgilia — APPROVED** [review]

**Pull request merged**